ql/src/java/org/apache/hadoop/hive/ql/security/authorization/plugin/metastore/events/CreateTableEvent.java

```diff
@@ -79,6 +83,24 @@ private List<HivePrivilegeObject> getOutputHObjs() {
     Database                  database = event.getDatabase();
     String                    uri   = getSdLocation(table.getSd());
 
+    // if the table is an iceberg storagehandler based table , need storageuri as hive privilege objects
+    // format: storagehandler type + cluster + location
+    LOG.debug("<== CreateTableEvent.getOutputHObjs(): ret={}", ret);
+    if (table.getParameters().containsKey(hive_metastoreConstants.META_TABLE_STORAGE)) {
+      Configuration conf = new Configuration();
+      try {
+        HiveStorageHandler hiveStorageHandler = (HiveStorageHandler) ReflectionUtils.newInstance(
+                conf.getClassByName(table.getParameters().get(hive_metastoreConstants.META_TABLE_STORAGE)), event.getHandler().getConf());
+        String storageUri = hiveStorageHandler.getURIForAuth(table).toString();
+        if (storageUri.contains("iceberg")) {
```

Comment:
Why do you need to check for iceberg? 
This code looks similar to `AlterTableEvent#getOutputHObjs`

Reply:
The reason we checked for iceberg is that a user using iceberg storage handler to create a new table is the only case this code patch would handle.
Yes, it looks similar to AlterTableEvent#getOutputHObjs because we need to do the same process for both create and alter iceberg storage handler based table.

Reply:
@DanielZhu58, why don't we have 'iceberg' check in AlterTableEvent? should we add it?

Reply:
It's also not clear for me why we only Authorize URI for iceberg only when others - hbase, kafka, jdbc - could have similar URI policies to be authorized (although URIs are computed in different way for each types)